packages/kinvey-datastore/src/datastores/datastore.js

```diff
@@ -5,8 +5,8 @@ import { isDefined } from 'kinvey-utils';
 import NetworkStore from './networkstore';
 import CacheStore from './cachestore';
 import SyncStore from './syncstore';
-// import { processorFactory } from '../processors';
-// import { repositoryProvider } from '../repositories';
+import { processorFactory } from '../processors';
+import { repositoryProvider } from '../repositories';
```

Comment:
The `DataStore` class isn't a data store in the sense that `SyncStore`, `CacheStore` and `NetworkStore` are, they do not inherit it. It's merely a factory for them.

Turns out, the `DataStore` class has a static `clear()` method, which deletes all data. The reason I can't instantiate, say, a `SyncStore` and clear its data, is that it clears data only for the collection it is created for and it doesn't know what other collections exist. I can't think of a better way of achieving this.

Reply:
I haven't dug into the details yet, but isn't there a way to delegate this to the processor instead?

Reply:
Processors also work on a "by collection" basis. The first class in the hierarchy, which knows what collections exist for a particular app key is the repository. We can, of course, do anything. The offline repositories know how to handle a clear when no collection is passed - we can have a separate processor method (this would require a separate operation type), or a setting in the options object of `clear()`, for the processor to know to call `clear()` without the collection parameter (hacky).

Why have it in a processor? Repositories are a data access layer and as such can be used by any non-public class. I don't think it's that bad to have it here. It's not worse than adding an operation type, in order to be able to instantiate a processor (since this is a static method on the `Kinvey.DataStore` object and no data store is instantiated at that point), just to call one of its methods.